source/Calamari.Tests/AppServiceBehaviorFixture.cs

```diff
@@ -217,6 +218,37 @@ public async Task DeployingWithInvalidEnvironment_ThrowsAnException()
                 commandResult.Outcome.Should().Be(TestExecutionOutcome.Unsuccessful);
             }
 
+            [Test]
+            public async Task DeployToTwoTargetsInParallel_Succeeds()
+            {
+                // Arrange
+                var packageInfo = PrepareFunctionAppZipPackage();
+                // Without larger changes to Calamari and the Test Framework, it's not possible to run two Calamari
+                // processes in parallel in the same test method. Simulate the file locking behaviour by directly
+                // opening the affected file instead
+                var fileLock = File.Open(packageInfo.packagePath, FileMode.Open, FileAccess.Read, FileShare.Read);
```

Comment:
I tried at first to do two `CommandTestBuilder.CreateAsync<DeployAzureAppServiceCommand, Program>()`, but this class intentionally doesn't spin up a full Calamari process, so that the Calamari code can be debugged during testing. It uses reflection to spin up the `CalamariProgramAsync` class. However, this class has some non-thread-safe access of `Environment.CurrentDirectory` which causes transient failures when done like this in a test context.

In a real deployment, each parallel target spins up a process-isolated instance of `Calamari.exe`, so we don't encounter that same problem. This seemed like way too big a thing to tackle here, so I've just simulated the problem by creating a lock on the target file.